File: sdk/typescript/src/api/investors.ts
import type { HttpClient, RequestOptions } from "../http.js";

export type InvestorEventType =
  | "LANDING_VIEW"
  | "DECK_VIEW"
  | "PAGE_VIEW"
  | "CTA_CLICK"
  | "LINK_CLICK"
  | "SCROLL_DEPTH"
  | "TIME_SPENT"
  | "CUSTOM";

export interface TrackInvestorEventBody {
  eventType: InvestorEventType;
  /** The specific page/section viewed (e.g. "hero", "product", "ask"). */
  page?: string;
  metadata?: Record<string, unknown>;
  /** Client-generated session identifier for grouping events. */
  sessionId?: string;
}

/**
 * Public investor deck and landing page access plus engagement event tracking.
 */
export class InvestorsApi {
  constructor(private readonly http: HttpClient) {}

  /** Get an investor deck by slug. */
  getInvestorPage<T = unknown>(slug: string, options?: RequestOptions): Promise<T> {
    return this.http.get<T>(`/investors/${encodeURIComponent(slug)}`, options);
  }

  /** Track an investor page event. */
  trackInvestorEvent<T = unknown>(
    slug: string,
    body: TrackInvestorEventBody,
    options?: RequestOptions,
  ): Promise<T> {
    return this.http.post<T>(
      `/investors/${encodeURIComponent(slug)}/events`,
      body,
      options,
    );
  }
}

File: sdk/rust/src/api/investors.rs
//! Public investor deck/landing page access plus engagement event tracking.

use reqwest::Method;
use serde_json::Value;

use crate::{enc, Error, HttpClient};

/// Typed client for the `/investors/*` routes.
pub struct InvestorsApi<'a> {
    http: &'a HttpClient,
}

impl<'a> InvestorsApi<'a> {
    pub fn new(http: &'a HttpClient) -> Self {
        Self { http }
    }

    /// Get an investor deck by slug.
    pub async fn get_investor_page(&self, slug: &str) -> Result<Value, Error> {
        let path = format!("/investors/{}", enc(slug));
        self.http.send(Method::GET, &path, &[], None, true).await
    }

    /// Track an investor page event.
    pub async fn track_investor_event(&self, slug: &str, body: &Value) -> Result<Value, Error> {
        let path = format!("/investors/{}/events", enc(slug));
        self.http.send(Method::POST, &path, &[], Some(body), true).await
    }
}
